akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala

```diff
@@ -19,15 +19,40 @@ import org.reactivestreams._
 
 import scala.concurrent.{ Await, ExecutionContextExecutor }
 
-object ActorFlowMaterializerImpl {
-  import OperationAttributes._
-  private[akka] def calcSettings(opAttr: OperationAttributes)(settings: ActorFlowMaterializerSettings): ActorFlowMaterializerSettings =
+/**
+ * INTERNAL API
+ */
+private[akka] object ActorFlowMaterializerImpl {
+
+  def calcSettings(opAttr: OperationAttributes)(settings: ActorFlowMaterializerSettings): ActorFlowMaterializerSettings = {
+    import OperationAttributes._
     opAttr.attributes.collect {
       case InputBuffer(initial, max) ⇒ (s: ActorFlowMaterializerSettings) ⇒ s.withInputBuffer(initial, max)
       case Dispatcher(dispatcher) ⇒ (s: ActorFlowMaterializerSettings) ⇒ s.withDispatcher(dispatcher)
       case SupervisionStrategy(decider) ⇒ (s: ActorFlowMaterializerSettings) ⇒
         s.withSupervisionStrategy(decider)
     }.reduceOption(_ andThen _).getOrElse((x: ActorFlowMaterializerSettings) ⇒ x)(settings) // FIXME is this the optimal way of encoding this?
+  }
+
+  /**
+   * Context parameter to the `create` methods of sources and sinks.
+   */
+  class CreateContext(
+    val effectiveAttributes: OperationAttributes,
+    val effectiveSettings: ActorFlowMaterializerSettings,
+    val stageName: String,
+    materializerImpl: ActorFlowMaterializerImpl) {
+
+    def materializer: ActorFlowMaterializer = materializerImpl
+
+    def actorOf(props: Props): ActorRef = {
+      val dispatcher =
+        if (props.dispatcher == Dispatchers.DefaultDispatcherId) effectiveSettings.dispatcher
```

Comment:
I thought so too, but actually not:

    def dispatcher: String = deploy.dispatcher match {
      case NoDispatcherGiven ⇒ Dispatchers.DefaultDispatcherId
      case x                 ⇒ x
    }